chore(devDeps): bump metro and react-native

legobeat opened the pull request:
## **Description**

Update react-native and metro to receive patches.
- Update metro from `0.73` to `0.76` 
- Migrate from `metro-config` to `@react-native/metro-config`
- Update `react-native` from `0.71.15` to `0.72.14`
  - Port patch
  - Remove hermes patch
- Bump and unpin `@react-native-community/` packages
- Update `@react-native-community/netinfo` from `6.0.0` to `^9.5.0`
  - [Changelog](https://github.com/react-native-netinfo/react-native-netinfo/blob/master/CHANGELOG.md)
  - [Release notes](https://github.com/react-native-netinfo/react-native-netinfo/releases)
  - v10 breaking change:
    > netinfo now requires AbortController
- Update `@react-native-masked-view/masked-view`
  - [Release notes](https://github.com/react-native-masked-view/masked-view/releases) 
- Upgrade to Gradle version 8.0.1 and Android SDK 33


React Native [Upgrade](https://github.com/orgs/MetaMask/projects/60/views/6?pane=issue&itemId=41033345)
[Android SDK Update](https://github.com/react-native-community/cli/issues/1856)

## **Related issues**


- Fixes: https://github.com/MetaMask/mobile-planning/issues/1262
- #7759

## **Manual testing steps**

Node environment must be `18.18.2`
Supports Java version `17.0.10`

1. Check out branch
2. Run `yarn setup`
3. Run `yarn watch`
4. Run `yarn start:ios` - wait for app to build on iOS simulator
5. Run `yarn start:android` - wait for app to build Android emulator


## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

legobeat committed:
chore(devDeps): bump metro from 0.73 to 0.76

legobeat committed:
devDeps: bump react-native from 0.71.15 to 0.72.14

legobeat committed:
port react-native patch

legobeat committed:
remove react-native hermes patch

legobeat committed:
chore: bump @react-native-community packages

legobeat committed:
deps: @react-native-community/netinfo@6.0.0->^9.5.0

legobeat committed:
deps: @react-native-masked-view/masked-view@^0.2.6->^0.3.1

legobeat committed:
deps: react-native-reanimated@3.1.0->^3.7.2

legobeat committed:
update fixes

leotm committed:
update react-native patch

leotm commented on patches/react-native+0.72.12.patch:
```diff
@@ -15,7 +15,8 @@ index 1379ffd..f2825f5 100644
  require('./setUpGlobals');
-+// require('./setUpDOM'); Introduced in RN v0.72, ensure included when upgrading patch
+-require('./setUpDOM');
++// require('./setUpDOM');
```
good spot ^ yep can confirm it's present [upstream](https://github.com/facebook/react-native/blob/v0.72.12/packages/react-native/Libraries/Core/InitializeCore.js#L30) so we want it uncommented

leotm replied:
updated our patch here https://github.com/MetaMask/metamask-mobile/pull/9280/commits/4b499b629d174e3e5aa45bbcce72257a562dfcbf

leotm commented on package.json:
```diff
@@ -252,7 +252,7 @@
     "randomfill": "^1.0.4",
     "react": "18.2.0",
-    "react-native": "0.71.15",
+    "react-native": "0.72.12",
```
spotted RN [v0.72.13](https://github.com/facebook/react-native/releases/tag/v0.72.13) release we may want to bump to while at it

leotm replied:
we'll want to make these changes too https://react-native-community.github.io/upgrade-helper/?from=0.71.15&to=0.72.13&package=io.metamask&name=MetaMask cc @MetaMask/mobile-platform

leotm commented on patches/react-native+0.72.12.patch:
tryna figure why getting these changes line 131 onwards (not picked up in https://github.com/MetaMask/metamask-mobile/pull/9280/commits/e1d12c8b3f562c636f44c5268b39ef159fabc438 https://github.com/MetaMask/metamask-mobile/pull/9280/commits/92fb6f983d9ee15cdee8d5a34a2078d482c7538f)
or if it's something that happens on Macs
or whether it's just patch-package _doing it's thing_

leotm commented on patches/react-native+0.72.12.patch:
```diff
@@ -122,43 +122,16 @@ index d03f3dd..816f559 100644
          minSdkVersion(21)
 diff --git a/node_modules/react-native/ReactAndroid/hermes-engine/build.gradle b/node_modules/react-native/ReactAndroid/hermes-engine/build.gradle
-index c714f87..65f9ba3 100644
+index 5ebaf48..f5dde30 100644
 --- a/node_modules/react-native/ReactAndroid/hermes-engine/build.gradle
 +++ b/node_modules/react-native/ReactAndroid/hermes-engine/build.gradle
-@@ -43,11 +43,11 @@ def overrideHermesDir = System.getenv("REACT_NATIVE_OVERRIDE_HERMES_DIR") != nul
- def hermesDir = System.getenv("REACT_NATIVE_OVERRIDE_HERMES_DIR") ?: new File(reactNativeRootDir, "sdks/hermes")
- def hermesBuildDir = new File("$buildDir/hermes")
- 
--def hermesVersion = "main"
--def hermesVersionFile = new File(reactNativeRootDir, "sdks/.hermesversion")
--if (hermesVersionFile.exists()) {
--    hermesVersion = hermesVersionFile.text
--}
-+def hermesVersion = "52725e889185e53277e629f1f7043aa2913c97a9"
-+// def hermesVersionFile = new File(reactNativeRootDir, "sdks/.hermesversion")
-+// if (hermesVersionFile.exists()) {
-+//     hermesVersion = hermesVersionFile.text
-+// }
- def ndkBuildJobs = Runtime.runtime.availableProcessors().toString()
- def prefabHeadersDir = new File("$buildDir/prefab-headers")
- 
-@@ -59,7 +59,7 @@ def skipPrefabPublishing = System.getenv("REACT_NATIVE_HERMES_SKIP_PREFAB") != n
- def jsiDir = new File(reactNativeRootDir, "ReactCommon/jsi")
- 
- task downloadHermes(type: Download) {
--    src("https://github.com/facebook/hermes/tarball/${hermesVersion}")
-+    src("https://github.com/MetaMask/hermes/tarball/${hermesVersion}")
-     onlyIfNewer(true)
-     overwrite(false)
-     dest(new File(downloadsDir, "hermes.tar.gz"))
-@@ -120,12 +120,9 @@ android {
```
just noting here removing our `node_modules/react-native/ReactAndroid/hermes-engine/build.gradle` patch to match upstream, should be reverted along with https://reactnative.dev/contributing/how-to-build-from-source#update-your-project-to-build-from-source to switch back to bundled Hermes cc @Cal-L @jpcloureiro

(original convo [here](https://github.com/MetaMask/metamask-mobile/pull/6220#discussion_r1218890123) and more context [here](https://github.com/facebook/hermes/issues/1241))

then if we need to build from source again, current simpler way is with [`REACT_NATIVE_OVERRIDE_HERMES_DIR`](https://github.com/facebook/hermes/blob/main/doc/ReactNativeIntegration.md)

leotm replied:
done in https://github.com/MetaMask/metamask-mobile/pull/9280/commits/40a238dfaf553e036be5c2e1450f16df2a9a6d85 and https://github.com/MetaMask/metamask-mobile/commit/6266824ea170ffc046db15d900d274a74cecf0c8

leotm reviewed (commented):
i did a brief check of [reactnative.dev/docs/0.71/environment-setup](https://reactnative.dev/docs/0.71/environment-setup) vs [reactnative.dev/docs/0.72/environment-setup](https://reactnative.dev/docs/0.72/environment-setup)

couldn't spot anything we need to update in our [docs](https://github.com/MetaMask/metamask-mobile/blob/main/docs/readme/environment.md), but may have missed something cc @MetaMask/mobile-platform

nb: JDK 17 is required in RN v0.73+
nb: Android 14 (UpsideDownCake) SDK required in RN v0.73+

nb: Node 18+ required in RN v0.73+ (we're ahead on Node 20)

nb: babel.config.js: [metro-react-native-babel-preset](https://github.com/facebook/metro/tree/v0.77.0/packages/metro-react-native-babel-preset) -> [@react-native/babel-preset](https://github.com/facebook/react-native/tree/main/packages/react-native-babel-preset) in RN v0.73+

nb: RN new experimental [debugger](https://reactnative.dev/docs/0.73/debugging?js-debugger=new-debugger) preview RN v0.73+ to replace Flipper

Daniel-Cross committed:
initial commit

Daniel-Cross committed:
small fix for android

leotm commented on package.json:
```diff
@@ -443,9 +442,9 @@
-    "metro-config": "~0.73.10",
-    "metro-react-native-babel-preset": "~0.73.10",
-    "metro-react-native-babel-transformer": "~0.73.10",
+    "metro-config": "~0.76.9",
+    "metro-react-native-babel-preset": "~0.76.9",
+    "metro-react-native-babel-transformer": "~0.76.9",
```
we'll also want to update 

https://github.com/MetaMask/metamask-mobile/blob/5dc336c10b5c27f8ebddc3f059489620d6dc4aad/metro.config.js#L9-L14

to extend [`@react-native/metro-config`](https://github.com/facebook/react-native/blob/v0.72.13/packages/react-native/template/metro.config.js) instead

fresh `npx react-native@0.72.13 init RN07213 --version 0.72.13` warns

```console
warn From React Native 0.72, your metro.config.js file should extend'@react-native/metro-config'. Please see the React Native 0.72 changelog, or copy the template at:
https://github.com/facebook/react-native/blob/main/packages/react-native/template/metro.config.js
warn Falling back to internal defaults.
```

leotm replied:
https://react-native-community.github.io/upgrade-helper/?from=0.71.15&to=0.72.13&package=io.metamask&name=MetaMask#RnDiffApp-metro.config.js

leotm replied:
great Qu ^ it should be [rn-diff-purge](https://github.com/react-native-community/rn-diff-purge)'ing
`npx react-native@0.71.15 init RnDiffApp --version 0.71.15`
vs
`npx react-native@0.72.13 init RnDiffApp --version 0.72.13`
according to [react-native-community/upgrade-helper](https://github.com/react-native-community/upgrade-helper)

leotm replied:
(but it's named `HelloWorld` [upstream](https://github.com/facebook/react-native/blob/v0.72.13/packages/react-native/template/app.json) for some reason)

leotm replied:
ah yes knew it rings a bell ^ (noting it in our [TS 4.8.4 upg](https://github.com/MetaMask/metamask-mobile/pull/6080))

https://github.com/MetaMask/metamask-mobile/blob/444c0dfdad94553cd2575ea87b4bd7f7f81caf75/.eslintrc.js#L2-L9

> The one thing I can still see diffing in the upgrade helper link (whee is that manifest sourced from btw?) there is migrating from `@react-native-community/eslint-config` to `@react-native/eslint-config`. Trying that seems to break storybook, however, so I guess that would be postponed to an upcoming storybook upgrade.

imo postponing this eslint upg/migration till after this RN upg is merged should be ok and help keep both tasks smaller @brianacnguyen

legobeat committed:
chore(devDeps): metro-config@~0.76.9 -> @react-native/metro-config@^0.72.12

legobeat committed:
chore(devDeps): eslint-plugin-react@^7.16.0->^7.34.1

leotm commented:
just noting we've removed old https://github.com/facebook/react-native/blob/v0.71.15/template/_node-version
in RN 72 upstream https://github.com/facebook/react-native/tree/v0.72.14/packages/react-native/template
so we may want to remove ours too, currently here
https://github.com/MetaMask/metamask-mobile/blob/main/.nvmrc

legobeat commented:
> just noting we've removed old [.node-version](https://github.com/facebook/react-native/blob/v0.71.15/template/_node-version) in RN 72 [upstream](https://github.com/facebook/react-native/tree/v0.72.14/packages/react-native/template) so we may want to remove our [.nvmrc](https://github.com/MetaMask/metamask-mobile/blob/main/.nvmrc) too to simplify our setup if not causing any issues

It's still used in GitHub Actions CI.
But in any case, it would be nice to consolidate the declaration of the project nodejs version.

We could already tell `asdf` to read version from `.nvmrc` as well: https://github.com/asdf-vm/asdf-nodejs?tab=readme-ov-file#nvmrc-and-node-version-support

They do refer to it as legacy, though. An alternative would be to replace `.nvmrc` with the newer `.tool-versions`, which `@actions/setup-node` [also supports](https://github.com/actions/setup-node/blob/c2ac33f2c62f978d6c944d9648125a294e56dc0b/action.yml#L11).

legobeat committed:
Refresh Podfile.lock

legobeat committed:
chore: update eslintrc

legobeat committed:
test: remove forwardedRef from snapshots

legobeat committed:
test: some things are not collapsable anymore; update snapshots.

legobeat committed:
test: remove AnimatedComponentWrapper as ForwardedRef param

legobeat committed:
chore: fix double-escaped sourceJson string

legobeat committed:
fix(devDeps): replace @babel/plugin-proposal-numeric-separator with @babel/plugin-transform-numeric-separator

legobeat committed:
fixme: ignore lint

legobeat committed:
test: undefined is not same as empty string anymore

Daniel-Cross commented:
I'm looking into this again now the Hackathon is over. Managed to get the branch in sync with `main` and build on iOS but I get the following issue with @walletconnect:
![Simulator Screenshot - iPhone 15 Pro - 2024-05-13 at 16 28 00](https://github.com/MetaMask/metamask-mobile/assets/34304333/cda731a5-f7b7-424e-9601-7cfc33bd3321)

Does anyone have any idea as to why this is occurring or any context for the issue? I saw there's been some recent changes to the package in the last week or so.

legobeat committed:
Merge branch 'main' into devdeps-metro

Commit:
working and compliing versions on both os

Commit:
updated lock screen snap file

Daniel-Cross committed:
merged in snapshots

Daniel-Cross committed:
merged in snapshots

Daniel-Cross committed:
merged in last couple of commits to the main branch

Daniel-Cross committed:
merged in last couple of commits to the main branch

Daniel-Cross committed:
Merge branch 'main' into update/react-native-upgrade-version-73x

Daniel-Cross committed:
Merge branch 'main' into update/react-native-upgrade-version-73x

Daniel-Cross committed:
updated podfile

Daniel-Cross committed:
updated podfile

Daniel-Cross committed:
snap update

Daniel-Cross committed:
snap update

Daniel-Cross committed:
merging in RN changes that allow the app to build and complie

Daniel-Cross committed:
merging in RN changes that allow the app to build and complie

Daniel-Cross committed:
fixed conflicts from main

Daniel-Cross committed:
fixed conflicts from main

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

legobeat commented:
@Daniel-Cross needs a `yarn deduplicate`

Daniel-Cross committed:
fixed build error

Daniel-Cross committed:
fixed build error

Daniel-Cross committed:
yarn deduplicate

Daniel-Cross committed:
yarn deduplicate

Daniel-Cross committed:
put back e2e exmaple env

Daniel-Cross committed:
put back e2e exmaple env

Daniel-Cross committed:
merged main into branch

Daniel-Cross committed:
merged main into branch

legobeat commented:
@SocketSecurity ignore npm/is-async-function@2.0.0
    @SocketSecurity ignore npm/@react-native-community/eslint-plugin@1.3.0
    @SocketSecurity ignore npm/@react-native/js-polyfills@0.72.1
    @SocketSecurity ignore npm/babel-plugin-transform-flow-enums@0.0.2

new authors ok

legobeat commented:
@SocketSecurity ignore npm/@react-native-community/cli-doctor@11.4.1

shell access ok

legobeat committed:
Merge branch 'main' into devdeps-metro

legobeat committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
merged main

Daniel-Cross committed:
merged main

legobeat commented:
> @Daniel-Cross needs a `yarn deduplicate`

Again a `yarn dedupe` should clear the failing ci check  (:

Cal-L committed:
Disable segment in E2E mode

Daniel-Cross committed:
merged in e2e fix

Daniel-Cross committed:
merged in e2e fix

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
merged latest main changes

Daniel-Cross committed:
merged latest main changes

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
made some changes to android build files

Daniel-Cross committed:
made some changes to android build files

legobeat committed:
Merge branch 'main' into devdeps-metro

legobeat committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
removed an exclusion on the build command for Detox on Android

Daniel-Cross committed:
removed an exclusion on the build command for Detox on Android

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
ran yarn deduplicate

Daniel-Cross committed:
ran yarn deduplicate

Daniel-Cross committed:
fixed conflicts

Daniel-Cross committed:
fixed conflicts

Daniel-Cross committed:
merged main and fixed conflicts

Daniel-Cross committed:
merged main and fixed conflicts

Daniel-Cross committed:
updated package.json and removed two resolutions no longer needed

Daniel-Cross committed:
updated package.json and removed two resolutions no longer needed

Daniel-Cross committed:
ran yarn deduplicate

Daniel-Cross committed:
ran yarn deduplicate

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

legobeat committed:
Merge branch 'main' into devdeps-metro

legobeat committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
merged main

Daniel-Cross committed:
merged main

Daniel-Cross committed:
ran yarn deduplicate

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
working and compliing versions on both os

Daniel-Cross committed:
updated lock screen snap file

leotm committed:
Merge remote-tracking branch 'origin/devdeps-metro' into devdeps-metro

leotm committed:
yarn deduplicate

leotm committed:
Update yarn.lock with @walletconnect

leotm committed:
Fix yarn audit and cleanup

leotm committed:
Update yarn lockfile

Daniel-Cross committed:
snapshots updated

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
Revert "Update yarn lockfile"

This reverts commit ea8258d531e446bf345b65bfd34fa29a08fd76a3.

Daniel-Cross committed:
undo revert

Daniel-Cross committed:
Revert "snapshots updated"

This reverts commit 33ea1f8c41cfe2f8e0d9604f1e4b04bbe873c532.

Daniel-Cross committed:
changed android emulator for e2e tests back to the one used on main

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
bumped to 0.72.15 to include a privacy manifest fix

Daniel-Cross committed:
resolved conflicts

Daniel-Cross committed:
removed 2 dependencies to be inline with main

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
ran yarn deduplicate

AbuSabreen commented:
added14b41e252696fa4f6e5a6f63b872d00e449
https://t.me/elhadel_coin/23

leotm committed:
Update min. iOS ver supported from '12.4' to min_ios_version_supported

Was '12.4' hard-coded in RN 0.70
https://github.com/facebook/react-native/blob/v0.70.15/template/ios/Podfile#L4

Use min_ios_version_supported var since RN 0.71
https://github.com/facebook/react-native/blob/v0.71.0/template/ios/Podfile#L4

Still '12.4' internally in RN 0.72
https://github.com/facebook/react-native/blob/0.72-stable/packages/react-native/scripts/cocoapods/__tests__/codegen_utils-test.rb#L23

Later '13.4' internally in RN 0.73
https://github.com/facebook/react-native/blob/0.73-stable/packages/react-native/scripts/cocoapods/__tests__/codegen_utils-test.rb#L23

Daniel-Cross committed:
updated snapshots

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
changed terms modal to be less expensive

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
changed ocnditional render

Daniel-Cross committed:
updated snapshot

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Cal-L committed:
chore: Chore/9924 - Enable `@typescript-eslint/no-explicit-any` as error rule (#10048)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

This PR adds the `@typescript-eslint/no-explicit-any` eslint rule to
prevent the usage of `any`. Aside from adding the rule to eslintrc, all
of the changes ignores the rule for existing usage of `any`

This is the mobile equivalent of
https://github.com/MetaMask/metamask-extension/pull/23531/ for extension

## **Related issues**

Fixes: #9924 

## **Manual testing steps**

- Run `yarn lint`
- Should not see errors about `@typescript-eslint/no-explicit-any`

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->
No lint errors
<img width="633" alt="image"
src="https://github.com/MetaMask/metamask-mobile/assets/10508597/420cb39a-826f-4d46-8a07-12f71121c102">


## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

OGPoyraz committed:
feat: enable transaction simulations (#10039)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

This PR enables transaction simulation in mobile which is implemented in
the https://github.com/MetaMask/metamask-mobile/pull/9828

## **Related issues**

This is the final task for mobile transaction simulation to enable it in
mobile.

## **Manual testing steps**

1. Either go for the simple send or contract interaction
2. See transaction simulations appear in the confirmation screen as
loading state first - then simulates the transaction

## **Screenshots/Recordings**

Recording could be found in the original implementation PR:
https://github.com/MetaMask/metamask-mobile/pull/9828

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [X] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [X] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

Co-authored-by: LeoTM <1881059+leotm@users.noreply.github.com>

OGPoyraz committed:
fix: hide amount in simulations for testnets if its opt out (#9918)

## **Description**


This PR aims to hide testnet fiat values if user opt out in the
settings.

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/25167?quickstart=1)

## **Related issues**

- Fixes: https://github.com/MetaMask/metamask-mobile/issues/9906

## **Manual testing steps**

1. Change current network to any testnet
2. Go to setting, turn on "Show conversion on test networks"
3. Try simple send - see fiat values displayed in simulations
4. Go to setting, turn off "Show conversion on test networks"
5. Try simple send - see no fiat values displayed in simulations
6. Change current network to mainnet or any other network which is not
testnet
7. Try simple send - see fiat values displayed in simulations


## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [X] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [X] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [X] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

tommasini committed:
fix: Auto capitalize letter to none to have the same behaviour as the password field on create new wallet flow (#10034)

## **Description**
Auto capitalize letter to none to have the same behaviour as the
password field on create new wallet flow

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

omridan159 committed:
fix: deeplink is not opening the site in the inapp-browser when the wallet is locked (#9843)

This PR addresses the issue where deeplinks were not opening the site in
the in-app browser when the wallet was locked.

Solution:
The logic handling deeplinks has been moved from running after the
`appTriggeredAuth` function to the after `postInit` method of the
SDKConnect instance.
This ensures that the deeplink functions are executed only after the
user has logged in and unlocked the wallet.

This is a Fix to this issues =>
https://github.com/MetaMask/metamask-mobile/issues/9632

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Related issues**

Fixes: #9632 

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**


https://github.com/MetaMask/metamask-mobile/assets/61094771/7aec8835-37e1-45e9-a3e9-e4c50de65913


<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

sethkfman committed:
chore: add typescript fitness function (#9960)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Related issues**

Fixes:

## **Manual testing steps**

1. Action PASS [Test link to add JS
file](https://github.com/MetaMask/metamask-mobile/actions/runs/9492883203/job/26160731282)
2. Action FAIL [Test link to modified JS
file](https://github.com/MetaMask/metamask-mobile/actions/runs/9492701540)
3. Action PASS [Test link to add TS
file](https://github.com/MetaMask/metamask-mobile/actions/runs/9492905460)

## **Screenshots/Recordings**

NA

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

Co-authored-by: Cal Leung <cleun007@gmail.com>
Co-authored-by: Cal Leung <cal.leung@consensys.net>

metamaskbot committed:
chore: New Crowdin translations by Github Action (#9777)

Co-authored-by: metamaskbot <metamaskbot@users.noreply.github.com>
Co-authored-by: sethkfman <seth.kaufman@consensys.net>

jpuri committed:
fix: improve message display for typed sign messages (#9795)

sahar-fehri committed:
chore: add smokeAssets e2e tag (#10067)

## **Description**

Adds a new assets E2E tag called SmokeAssets.


## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

SamuelSalas committed:
test: Fix detox test run inconsistencies (#10055)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

Lately, some of the test are failing both locally and on Bitrise. Most
of this errors are regarding some inconsistencies found on the
viewHelper.js file. Steps closing onboarding modals are being call
multiple times. So the best solution would be unifying all these steps
in just one method, so any change to modals can be just change in one
place and affect all test. As well, we need to fix some assertions
parameters that are not following the correct definitions from the
jsdocs and update the deprecated methods with their current
implementations.

## **Related issues**

Fixes: 

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

Smoke test run:
https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/9129dcc3-e46d-4572-b627-9f90c3feab96
Regression Test run: 

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

Daniel-Cross committed:
fixed conflicts

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
downgraded reanimated to 3.4.1

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
removed temp modal fix

AbuSabreen reviewed (commented):
لا اعرف ماذا افعل

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
changed detox version

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
made some changes to android files

Daniel-Cross committed:
merged main and fixed conflicts

Daniel-Cross committed:
updated yarn.lock

Daniel-Cross committed:
fixed duplicate issue in podfile

Daniel-Cross committed:
removed dattimepicker as it's not used anywhere in the codebase

Daniel-Cross committed:
patched @metamask/controller-utils with a catch block to stop Android crashing when swap value was higher than token amount

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
updated patch for controller

Daniel-Cross committed:
merged main and removed datetimepicker

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
updated snapshots and RN patch

Daniel-Cross committed:
added packagingOptions to build gradle

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
added patch for controller-utils

Daniel-Cross committed:
fix merge conflicts

Daniel-Cross committed:
yarn lock fix

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
merge and fix conflicts

Daniel-Cross committed:
yarn lock update

Daniel-Cross committed:
fixed merge conflicts

tommasini committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
removed patch

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
upgrade reanimated

FrederikBolding committed:
feat: Revamp Snap connection screen (#10189)

## **Description**

This PR reworks the Snaps connection screen to add the Snap icon and
simplify the implementation. It also tweaks the fencing to allow
connecting to Snaps with only the `preinstalled-snaps` flag on.

## **Screenshots/Recordings**

<img
src="https://github.com/MetaMask/metamask-mobile/assets/1561200/0a4a2efb-c3f7-48b0-a2c6-82b9d08489eb"
width="400" />

Daniel-Cross committed:
changed gradle files

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
merged main

Daniel-Cross committed:
updated yarn lock and podfile

Daniel-Cross committed:
removed unused package

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
copied over yarn lock from main

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Prithpal-Sooriya committed:
fix: add support for svg uris without viewbox (#10247)

## **Description**

We have ran into some SVG rendering issues with SVG URIs. This is due to
some SVGs do not contain a view-box, thus can be much larger or smaller
than the container it is trying to fit in.

For example:
- [This
SVG](https://raw.githubusercontent.com/MetaMask/contract-metadata/master/images/usdt.svg)
does not have a viewbox, and does not "render correctly"
- [This
SVG](https://token.api.cx.metamask.io/assets/nativeCurrencyLogos/ethereum.svg)
does have a viewbox and can be rendered correctly.

<details><summary>See Example Screenshot</summary>

![Screenshot 2024-07-04 at 02 36
32](https://github.com/MetaMask/metamask-mobile/assets/17534261/71b29017-bb0c-43a0-8b36-eba93489c643)

</details> 

This adds an effect to use the SVG width/height for setting a view box
if not set in the SVG.

## **Related issues**

Known issue and temporary fix in `react-native-svg`:

https://github.com/software-mansion/react-native-svg/issues/1202#issuecomment-1891110599

## **Manual testing steps**


## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

<details><summary>Before</summary>

![Screenshot 2024-07-04 at 02 36
32](https://github.com/MetaMask/metamask-mobile/assets/17534261/1a716d94-cd63-4aad-a995-4075cc14346c)

</details> 

### **After**

<details><summary>After</summary>

![Screenshot 2024-07-04 at 12 32
35](https://github.com/MetaMask/metamask-mobile/assets/17534261/831cb931-01fc-4ffe-a4cc-1ef122c2f525)

</details> 

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

jpuri committed:
fix: copy changes in blockaid settings (#10233)

tommasini committed:
chore: Convert initial background state to typed background state (#9441)

## **Description**

A small set forward into an application with better typescript coverage.
Convert all files to use backgroundState from initial-root-state file
instead of the initial-background-state json file (unit test files)

## **Related issues**

Fixes: https://github.com/MetaMask/mobile-planning/issues/1734

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

wachunei committed:
chore(ramp): upgrade sdk to 1.28.1 (#9652)

metamaskbot committed:
chore: New Crowdin translations by Github Action (#10071)

Co-authored-by: metamaskbot <metamaskbot@users.noreply.github.com>
Co-authored-by: sethkfman <seth.kaufman@consensys.net>

Commit:
chore: Migrate AsyncStorage to mmkv (#10153)

## **Description**

While debugging a sqlite error related to AsyncStorage (surpassing
storage limit on Android) we have decided to migrate from AsyncStorage
to mmkv due to it not having said issue and also for performance
reasons. This includes an actual migration of any existing items saved
in AsyncStorage to mmkv

## **Related issues**

Fixes:
https://github.com/orgs/MetaMask/projects/60/views/6?pane=issue&itemId=63268337

## **Manual testing steps**

1. Start a fresh install
2. Create new wallet and set password, etc
3. Reboot app and see if it gives you normal `EXISTING_USER` flow (since
that value is taken from mmkv now)

## **Screenshots/Recordings**

Should be no discernible different in application behavior.

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [X] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

Co-authored-by: tommasini <46944231+tommasini@users.noreply.github.com>

github-actions[bot] committed:
7.26.0 (#10052)

This is the release candidate for version 7.26.0.

---------

Co-authored-by: metamaskbot <metamaskbot@users.noreply.github.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: OGPoyraz <omergoktugpoyraz@gmail.com>
Co-authored-by: LeoTM <1881059+leotm@users.noreply.github.com>
Co-authored-by: Nico MASSART <NicolasMassart@users.noreply.github.com>
Co-authored-by: sethkfman <seth.kaufman@consensys.net>
Co-authored-by: SamuelSalas <samuel.salas.reyes@gmail.com>
Co-authored-by: hesterbruikman <hester.bruikman@consensys.net>
Co-authored-by: EtherWizard33 <eric.lamontagne@consensys.net>
Co-authored-by: EtherWizard33 <165834542+EtherWizard33@users.noreply.github.com>
Co-authored-by: sethkfman <10342624+sethkfman@users.noreply.github.com>
Co-authored-by: tommasini <46944231+tommasini@users.noreply.github.com>
Co-authored-by: Cal Leung <cleun007@gmail.com>
Co-authored-by: Omri Dan <61094771+omridan159@users.noreply.github.com>
Co-authored-by: Christopher Ferreira <104831203+christopherferreira9@users.noreply.github.com>
Co-authored-by: Daniel <80175477+dan437@users.noreply.github.com>
Co-authored-by: yande <110056475+Andepande@users.noreply.github.com>
Co-authored-by: Derek Brans <dbrans@gmail.com>
Co-authored-by: Jony Bursztyn <jonathan@bursztyn.io>
Co-authored-by: Curtis David <Curtis.David7@gmail.com>
Co-authored-by: salimtb <salim.toubal@outlook.com>
Co-authored-by: Wietze Bronkema <11412480+wtzb@users.noreply.github.com>
Co-authored-by: Owen Craston <owen.craston@consensys.net>
Co-authored-by: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Co-authored-by: Cal Leung <cal.leung@consensys.net>
Co-authored-by: Mark Stacey <markjstacey@gmail.com>
Co-authored-by: Vinicius Stevam <45455812+vinistevam@users.noreply.github.com>
Co-authored-by: infiniteflower <139582705+infiniteflower@users.noreply.github.com>
Co-authored-by: sahar-fehri <sahar.fehri@consensys.net>

Daniel-Cross committed:
merged main and fixed conflicts

Daniel-Cross committed:
added patch for gas fee estimate

Daniel-Cross committed:
added extra case to if statement if gasfees is set to an empty object

Daniel-Cross committed:
added patch to permission controller for when user rejects request and data is undefined

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
Merge branch 'main' into devdeps-metro

Daniel-Cross committed:
added patch for reject function on approval controller

tommasini commented:
@jpcloureiro quote:

"
We had silent crashed when we [upgrade react native to 0.71.6](https://github.com/MetaMask/metamask-mobile/pull/6220)
The hermes patch resolved the silent crashes.
These crashes happened when the user rejected a wallet transaction

As stated on the PR description, we cherry-picked [one hermes commit](https://github.com/facebook/hermes/commit/f2f1a2e38e86632f24176d9e607c06507f900496) that wasn't included in the react-native project at that time.
The request to include it was reject as the react-native team was encountering issues with it.
https://github.com/facebook/hermes/commit/f2f1a2e38e86632f24176d9e607c06507f900496
https://github.com/reactwg/react-native-releases/discussions/69#discussioncomment-5861539


These are all the hermes version tags that include that fix
hermes-2024-07-01-RNv0.75.0-1edbe36ce92fef2c4d427f5c4e104f2758f4b692  
hermes-2024-06-28-RNv0.74.3-7bda0c267e76d11b68a585f84cfdd65000babf85 hermes-2024-06-17-RNv0.75.0-c2c4ee7dfcf4b006268ee215b63f79e9cbeedcf4 
hermes-2024-06-03-RNv0.74.2-bb1e74fe1e95c2b5a2f4f9311152da052badc2bc hermes-2024-04-29-RNv0.73.8-644c8be78af1eae7c138fa4093fb87f0f4f8db85 
hermes-2024-04-25-RNv0.74.1-b54a3a01c531f4f5f1904cb0770033e8b7153dff hermes-2024-04-25-RNv0.73.8-e3f36ed20b0d9db08e2f3723af2a0be080778f1e 
hermes-2024-03-26-RNvmain-86d391d00be80aeccc8b29f62a254ecf6bc8a9ae hermes-2024-02-20-RNv0.74.0-999cfd9979b5f57b1269119679ab8cdf60897de9 
hermes-2024-02-20-RNv0.73.5-18f99ace4213052c5e7cdbcd39ee9766cd5df7e4 hermes-2024-01-25-RNv0.73.3-57e3665290d5e97eaea63d27cb370ce7862a10a6 
hermes-2023-11-17-RNv0.73.0-21043a3fc062be445e56a2c10ecd8be028dd9cc5 
hermes-2023-09-26-RNv0.73.0-ee2922a50fb719bdb378025d95dbd32ad93cd679

> hey @leotm a quick update on this: we tried to cherry pick it but we hit a number of issues so we'll hold off from making this Hermes change.

[Source](https://github.com/reactwg/react-native-releases/discussions/69#discussioncomment-5861539)
I don't know specifically what issues the react-native team encountered when they tried to cherry-pick.
I do remember running hermes unit test locally and failing a couple of tests.
I would leave the hermes patch as last resort.
"

tommasini commented:
I Will start removing the commits that were fixing crashes and begin the steps to update to 0.73.8

joaoloureirop commented:
https://github.com/MetaMask/metamask-mobile/pull/9280#issuecomment-2231126228

This convo is around Android crashes we're observing on this PR. 

@tommasini verified that the Android crash goes away after adding back the hermes patch

tommasini committed:
revert last patches to fix crashes, permission controller and swaps controller patch removed, approval controller patch change reverted, added a patch to rpc errors, to fix the extended error class that was adding the stack of the error

tommasini commented:
After realizing that the hermes patch do not fix the issue!
It was identified that a loop was created with the stack of an error being added to a class the extends Error (JsonRpcError class)

Last patches fixing the crashes were reverted and it was added a patch to comment the code of the stack, added a comment to it to make sure we know why it's commented

Created a minimal reproduction for we to be able to see the issue easier:

Grabbing the stack property was creating a loop with hermes enabled
```
export default class CustomErrorr extends Error {
  constructor(code, message) {
    super(message);
    this.code = code;
  }

  serialize() {
    const serialized = {
      code: this.code,
      message: this.message,
    };
    serialized.stack = this.stack;

    return serialized;
  }

  toString() {
    console.log('ENTER CUSTOM ERROR TO STRING');
    return JSON.stringify(this.serialize());
  }
}
```

tommasini committed:
hermes ios enabled

tommasini committed:
revert hermes on ios

tommasini committed:
merge main and solve conflicts

tommasini committed:
Merge branch 'main' into devdeps-metro

tommasini commented:
@SocketSecurity ignore npm/@babel/plugin-proposal-numeric-separator@7.18.6

This is an acceptable risk, since it's part of the `metro-react-native-babel-preset` package

Cal-L reviewed (dismissed):
Nice work, LGTM! Since Dan is OOO, we can briefly disable CLA bot to get this merged. just LMK

georgewrmarshall reviewed (dismissed):
Approving on behalf of @MetaMask/design-system-engineers

joaoloureirop commented on .e2e.env.exmaple:
```diff
@@ -2,4 +2,4 @@
 export MM_TEST_ACCOUNT_PRIVATE_KEY=''
-export IS_TEST="true"
\ No newline at end of file
+export IS_TEST="true"
```
typo in this filename `exmaple`

legobeat committed:
chore: rename to .e2e.env.example

legobeat committed:
Merge branch 'main' into devdeps-metro

jpuri reviewed (dismissed):
Changes in files owned by confirmations team look great 👍

tommasini commented:
@legobeat Can we sync on it? I have a fix for that conversation locally but would like to understand why it was applied on the first place

wachunei reviewed (dismissed):
Ramp changes LGTM

leotm commented:
a wild `ios/Podfile.lock` conflict appeared ^ then lgtm

andreahaku reviewed (dismissed):
LGTM

tommasini committed:
podfile lock conflict solved

leotm commented on android/app/build.gradle:
```diff
@@ -193,6 +184,14 @@ android {
 
 	  packagingOptions {
 		  exclude 'META-INF/DEPENDENCIES'
+      pickFirst 'lib/x86/libc++_shared.so'
+      pickFirst 'lib/x86_64/libc++_shared.so'
+      pickFirst 'lib/armeabi-v7a/libc++_shared.so'
+      pickFirst 'lib/arm64-v8a/libc++_shared.so'
+      pickFirst 'lib/arm64-v8a/libcrypto.so'
+      pickFirst 'lib/armeabi-v7a/libcrypto.so'
+      pickFirst 'lib/x86/libcrypto.so'
+      pickFirst 'lib/x86_64/libcrypto.so'
```
are these still needed?

tommasini replied:
Yeah it seems that are still needed, it failes to build when removed
![image](https://github.com/user-attachments/assets/e42e37c7-85c5-4a1d-ac8a-dab326401e9c)

leotm commented on android/settings.gradle:
```diff
@@ -10,15 +10,8 @@ pluginManagement {
 rootProject.name = 'MetaMask'
 apply from: file("../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesSettingsGradle(settings)
 include ':app'
-includeBuild('../node_modules/react-native-gradle-plugin')
-includeBuild('../node_modules/react-native') {
-     dependencySubstitution {
-         substitute(module("com.facebook.react:react-android")).using(project(":ReactAndroid"))
-         substitute(module("com.facebook.react:react-native")).using(project(":ReactAndroid"))
-         substitute(module("com.facebook.react:hermes-android")).using(project(":ReactAndroid:hermes-engine"))
-         substitute(module("com.facebook.react:hermes-engine")).using(project(":ReactAndroid:hermes-engine"))
-     }
-}
```
nb: we can eventually remove https://github.com/MetaMask/hermes once fully no longer in use

tommasini committed:
revert metametrics changes

leotm reviewed (approved):
lgtm ^ thanks for reviewing with me @tommasini

legobeat committed:
Merge branch 'main' into devdeps-metro

legobeat committed:
Merge branch 'main' into devdeps-metro

Cal-L reviewed (approved):
LGTM

Cal-L commented:
Disabled CLA since Dan is OOO
